common/utils/include/claragenomics/utils/buffer.hpp

```diff
@@ -70,7 +70,7 @@ class buffer
     /// @param allocator The allocator to use by this buffer.
     /// @param stream The CUDA stream to be associated with this allocation. Default is stream 0.
     /// @tparam AllocatorIn Type of input allocator. If AllocatorIn::value_type is different than T AllocatorIn will be converted to Allocator<T> if possible, compilation will fail otherwise
-    template <typename AllocatorIn>
+    template <typename AllocatorIn, std::enable_if_t<std::is_class<AllocatorIn>::value, int> = 0>
```

Comment:
Why cannot it be union? A comment would help

Reply:
Yes, agreed, a comment would help here.
I don't think this can be a union, we assume that an Allocator provides `allocate()` and `deallocate()` functions and so on...

Reply:
done

Reply:
yes, makes sense